### PR TITLE
docs(skore): Use `evaluate` instead of creating reports manually

### PR DESCRIPTION
**Pull request by GaetandeCast**
#### Contribution checklist

- [ ] Unit tests were added or updated (if necessary)
- [ ] Documentation was added or updated (if necessary)
- [ ] The code passes our style conventions (you can check this by running pre-commit on your code
with `pre-commit run --all-files`)
- [ ] All the tests pass (please test locally before pushing)
- [ ] The documentation builds and renders properly (if it does, our bot will add a comment linking
to a preview of the documentation to review it visually)
- [ ] A new changelog entry was added to CHANGELOG.rst (if necessary; typically, if your change
requires updating tests)
- [ ] All the commits in the PR are signed (more information
[here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits))
- [ ] The pull request title respects the Conventional Commits convention (more information
[here](https://docs.skore.probabl.ai/stable/contributing.html#pull-requests))

#### AI usage disclosure

AI tools were involved for:
- [ ] Code generation (e.g., when writing an implementation or fixing a bug)
- [ ] Test/benchmark generation
- [ ] Documentation (including examples)
- [ ] Research and understanding

### CODE REVIEW

File: examples/getting_started/plot_getting_started.py
Comment (GaetandeCast) [inline examples/getting_started/plot_getting_started.py:243]: Note: I did not use `evaluate` here because we want to compare existing report while `evaluate` with multiple models would rebuild the reports. This raises the need for a `compare` function as was proposed by @glemaitre here: https://github.com/probabl-ai/skore/issues/2520#issuecomment-3987627477

File: sphinx/reference/index.rst [hunk 1]
```diff
@@ -30,6 +30,8 @@ ML Assistance
 .. list-table::
    :widths: 30 70
 
+   * - :func:`evaluate`
```
Comment (glemaitre) [inline sphinx/reference/index.rst:33]: Could you also add the `TrainTestSplit` splitter and mention that this is the splitter version compatible with the scikit-learn API (https://scikit-learn.org/stable/glossary.html#term-CV-splitter)

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 2]
```diff
@@ -116,19 +116,18 @@ def summarize(
...
-        >>> from skore import EstimatorReport
+        >>> from skore import evaluate
         >>> X, y = load_breast_cancer(return_X_y=True)
-        >>> split_data = train_test_split(X=X, y=y, random_state=0, as_dict=True)
-        ...
         >>> classifier = LogisticRegression(max_iter=10_000)
-        >>> report = EstimatorReport(classifier, **split_data, pos_label=1)
-        >>> report.metrics.summarize().frame(favorability=True)
+        >>> report = evaluate(classifier, X, y, splitter=0.2, pos_label=1)
+        >>> metrics_report = report.metrics.summarize().frame(favorability=True).frame(
```
Comment (glemaitre) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: Something is weird here. You have twice `.frame`

Comment (GaetandeCast) [reply skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: Good catch

File: skore/src/skore/_sklearn/_plot/inspection/coefficients.py [hunk 3]
```diff
@@ -49,29 +49,26 @@ class CoefficientsDisplay(DisplayMixin):
...
-    ...     X=X, y=y, random_state=0, as_dict=True, shuffle=True
-    ... )
-    >>> report = EstimatorReport(LogisticRegression(), **split_data)
+    >>> report = evaluate(LogisticRegression(), X, y, splitter=0.2)
     >>> display = report.inspection.coefficients()
     >>> display.frame()
                   feature       label  coefficient
     0           Intercept      setosa        9.2...
-    1   sepal length (cm)      setosa       -0.4...
+    1   sepal length (cm)      setosa       -0.3...
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_plot/inspection/coefficients.py:61]: Wait so `evaluate` no longer has `random_state`/`seed`?

Comment (auguste-probabl) [reply skore/src/skore/_sklearn/_plot/inspection/coefficients.py:61]: Oh it does, `random_state` is 0 by default:

https://github.com/probabl-ai/skore/blob/eddf218e4fb8ab872418a87eeff05d0b057736f9/skore/src/skore/_sklearn/train_test_split/train_test_split.py#L64

Comment (GaetandeCast) [reply skore/src/skore/_sklearn/_plot/inspection/coefficients.py:61]: Indeed, it is documented in the `splitter` parameter docstring.
